pandas/util/testing.py

```diff
@@ -1244,7 +1244,12 @@ def assert_series_equal(left, right, check_dtype=True,
                        obj='{obj}.index'.format(obj=obj))
 
     if check_dtype:
-        assert_attr_equal('dtype', left, right)
+        if (is_categorical_dtype(left) and is_categorical_dtype(right) and
+                not check_categorical):
+            # compat with pandas 0.21.0 CategoricalDtype
+            pass
```

Comment:
why is this needed?

Reply:
This test: https://github.com/pandas-dev/pandas/pull/16015/files#diff-5066b062c892df33c29b37a8c1c75f75L1093 

I don't recall the details at the moment...

Reply:
Ah, it seems this is rather logical at a second look, as if `check_categorical=False`, the dtypes of categorical should not be checked exactly. So I would just remove the comment? Or better clarify it?